experimental/report/renderer.go

```diff
@@ -261,34 +265,33 @@ func (r *renderer) diagnostic(report *Report, d Diagnostic) {
 		fmt.Fprintf(r, "--> %s", d.inFile)
 	}
 
-	// Render the footers. For simplicity we collect them into an array first.
-	footers := make([][3]string, 0, len(d.notes)+len(d.help)+len(d.debug))
-	for _, note := range d.notes {
-		footers = append(footers, [3]string{r.ss.bRemark, "note", note})
-	}
-	for _, help := range d.help {
-		footers = append(footers, [3]string{r.ss.bRemark, "help", help})
+	type footer struct {
+		color, label, text string
 	}
-	if r.ShowDebug {
-		for _, debug := range d.debug {
-			footers = append(footers, [3]string{r.ss.bError, "debug", debug})
+	footers := iterx.Chain(
+		slicesx.Map(d.notes, func(s string) footer { return footer{r.ss.bRemark, "note", s} }),
+		slicesx.Map(d.help, func(s string) footer { return footer{r.ss.bRemark, "help", s} }),
+		slicesx.Map(d.debug, func(s string) footer { return footer{r.ss.bError, "debug", s} }),
+	)
+
+	footers(func(f footer) bool {
+		isDebug := f.label == "debug"
+		if isDebug && !r.ShowDebug {
+			return true
 		}
-	}
-	for _, footer := range footers {
+
 		r.WriteString("\n")
-		r.WriteString(r.ss.nAccent)
 		r.WriteSpaces(r.margin)
-		r.WriteString(" = ")
-		fmt.Fprint(r, footer[0], footer[1], ": ", r.ss.reset)
-		for i, line := range strings.Split(footer[2], "\n") {
-			if i > 0 {
-				r.WriteString("\n")
-				margin := r.margin + 3 + len(footer[1]) + 2
-				r.WriteSpaces(margin)
-			}
-			r.WriteString(line)
+		fmt.Fprintf(r, "%s = %s%s: %s", r.ss.nAccent, f.color, f.label, r.ss.reset)
+
+		if isDebug {
+			r.WriteWrapped(f.text, math.MaxInt)
```

Comment:
Debug lines should not be wrapped.

Reply:
This code is correct. See comment.

Reply:
Hmmm, let me dig a little, since the test output seems to show wrapped debug lines...

Reply:
Oh lol, you got the fix -- resolving this!